### PR TITLE
Don't discard source on unavailable delta snapshot

### PR DESCRIPTION
**Pull request by grtlr**
# Description of change

Don't entirely discard a source if only the download of the delta snapshot fails.

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-ledger/src/workers/snapshot/download.rs [hunk 1]
```diff
@@ -126,7 +126,14 @@ async fn gather_source_information(
 ) -> Result<SourceInformation<'_>, Error> {
     let full_header = download_snapshot_header(urls.full()).await?;
     let delta_header = if download_delta {
-        Some(download_snapshot_header(urls.delta()).await?)
+        match download_snapshot_header(urls.delta()).await {
+            Ok(header) => Some(header),
+            Err(Error::DownloadingFailed) => {
+                debug!("Downloading delta snapshot header failed: {}", urls.delta());
```
Comment (thibault-martinez) [inline bee-ledger/src/workers/snapshot/download.rs]: But the previous warning is still there, right ?

Comment (samuel-rufi) [inline bee-ledger/src/workers/snapshot/download.rs]: we are doing error messages in lowercase, so I guess the same applies for debug messages

**Review by samuel-rufi — CHANGES_REQUESTED** [review]

File: bee-ledger/src/workers/snapshot/download.rs (hunk 1 above)
Comment (thibault-martinez) [reply bee-ledger/src/workers/snapshot/download.rs]: It doesn't!

Comment (thibault-martinez) [reply bee-ledger/src/workers/snapshot/download.rs]: Errors are not capitalised because they are included in another sentence, logs are not

Comment (samuel-rufi) [reply bee-ledger/src/workers/snapshot/download.rs]: oh makes sense, sorry!

**Review by thibault-martinez — COMMENTED** [review]
Can you draft a new patch version for the ledger crate please ?

**Review by thibault-martinez — APPROVED** [review]

File: bee-ledger/src/workers/snapshot/download.rs [hunk 2]
```diff
@@ -40,7 +40,7 @@ async fn download_snapshot_header(download_url: &str) -> Result<SnapshotHeader,
                 }
             }
         }
-        Err(e) => warn!("Downloading snapshot header failed: {:?}.", e.to_string()),
+        Err(e) => debug!("Downloading snapshot header failed: {:?}.", e.to_string()),
```
Comment (thibault-martinez) [inline bee-ledger/src/workers/snapshot/download.rs:43]: Actually wondering, should this be a warning for full and debug for delta ? Or do we just want to discover at the end that no source were available/working ?

Comment (grtlr) [reply bee-ledger/src/workers/snapshot/download.rs:43]: I would expect most users only care about the node starting correctly. So checking at the end should suffice.

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**